client/src/cmdhfmfp.c

```diff
@@ -1056,7 +1056,8 @@ static int MFPKeyCheck(uint8_t startSector, uint8_t endSector, uint8_t startKeyA
                 }
 
                 // 5 - auth error (rnd not equal)
-                if (res != 5) {
+                // PM3 client says that RND not equal is -16. Corrected. Seems to work.
+                if (res != -16) {
```

Comment:
Better to use the DEFINE's    Look in Auth4 function which return case you are dealing with.  
-16 ==  PM3_EWRONGANSWER